### PR TITLE
Share buttons, config with google-defs, fixes #48

### LINKED ISSUE #48: Social Share buttons
Add sharing functionality into Facebook and Twitter buttons on My Offerings and offering detail pages. Beware, when user wants to share My Offerings page, it actually should share his user offerings page: `/user/<useraddress>/offerings`., not `/my-offerings`

### PR DESCRIPTION
**Pull request by wambat**
I also added config build selector, to get rid of switching configs in the code every time i checkout. I personally find it usefull when deploying to several dev envs, like stage, dev, prod, local. But it's not strictly necessary for this PR.

### CODE REVIEW

File: src/name_bazaar/ui/components/share_buttons.cljs [hunk 1]
```diff
@@ -1,8 +1,30 @@
-(ns name-bazaar.ui.components.share-buttons)
+(ns name-bazaar.ui.components.share-buttons
```
Comment (madvas) [inline src/name_bazaar/ui/components/share_buttons.cljs]: Share buttons don't serve only to share My Offerings page, but also for offering detail page. Must do it general purpose components with passed props.

File: src/name_bazaar/ui/components/share_buttons.cljs [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+  (:require [re-frame.core :refer [reg-sub subscribe]]
+            [reagent.ratom :refer-macros [reaction]]
+            [clojure.string :as cstr]
```
Comment (madvas) [inline src/name_bazaar/ui/components/share_buttons.cljs]: Please use `:as string` for consistency with rest of app

File: src/name_bazaar/ui/components/share_buttons.cljs [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
 
 (defn share-buttons []
-  [:div.share-buttons
-   [:div.title "Share On:"]
-   [:i.icon.fb]
-   [:i.icon.twitter]])
+  (let [root-url (subscribe [:root-url])
+        my-address (subscribe [:district0x/active-address])
+        title "My ENS Offerings"
+        page-url (reaction (cstr/replace
```
Comment (madvas) [inline src/name_bazaar/ui/components/share_buttons.cljs]: Please use normal `subscribe` for consistency. Also you must check now if we're using HTML5 push state urls or with #, to share url correctly. See `district0x.ui.utils/hashroutes?`

File: src/name_bazaar/ui/db.cljs [hunk 4]
```diff
@@ -11,21 +11,28 @@
     [re-frame.core :refer [dispatch]]
     [district0x.ui.utils :as d0x-ui-utils]))
 
+(goog-define environment "production")
+
 (def development-config
   {:node-url "http://localhost:8549"
    :load-node-addresses? true
+   :root-url "https://beta.namebazaar.io"
```
Comment (madvas) [inline src/name_bazaar/ui/db.cljs]: Why not to detect this programmatically like from window.location object?

File: src/name_bazaar/ui/components/share_buttons.cljs (hunk 3 above)
Comment (madvas) [reply src/name_bazaar/ui/components/share_buttons.cljs]: Actually no need to detect this.`https://namebazaar.io` will always have urls without hashtag

File: src/name_bazaar/ui/db.cljs (hunk 4 above)
Comment (wambat) [reply src/name_bazaar/ui/db.cljs]: Just to make sure we have a canonical name in posts - probably window.location will be the same, but i didn't want to couple them this way.

File: src/name_bazaar/ui/components/share_buttons.cljs [hunk 5]
```diff
@@ -1,8 +1,14 @@
-(ns name-bazaar.ui.components.share-buttons)
+(ns name-bazaar.ui.components.share-buttons
+  (:require [re-frame.core :refer [reg-sub subscribe]]
+            [reagent.ratom :refer-macros [reaction]]))
 
-(defn share-buttons []
+(defn share-buttons [page-url title]
```
Comment (madvas) [inline src/name_bazaar/ui/components/share_buttons.cljs]: Please use it like props `:url` and `:title`

File: src/name_bazaar/ui/pages/user_offerings_page.cljs [hunk 6]
```diff
@@ -71,7 +71,9 @@
               :tablet 8
               :mobile 16
               :floated "right"}
-             [share-buttons]]]
+             [share-buttons
+              @(subscribe [:page-share-url :route.user/offerings {}])
+              "My ENS Offerings"]]]
```
Comment (madvas) [inline src/name_bazaar/ui/pages/user_offerings_page.cljs]: Actually, users can share other user's offerings page as well. So here should detect if it shares my offerings or a user offerings page and choose title accordingly.

File: src/name_bazaar/ui/subs.cljs [hunk 7]
```diff
@@ -42,3 +48,19 @@
...
+(reg-sub
+ :page-share-url
+ :<- [:root-url]
+ :<- [:district0x/active-address]
+ (fn [[root-url my-address] [_ route params]]
+   (string/replace
+    (str
+     root-url
+     "/"
+     (path-for route (merge {:user/address (if my-address
```
Comment (madvas) [inline src/name_bazaar/ui/subs.cljs]: Here again, this won't work if user wants to share other user's offerings page.

File: src/name_bazaar/ui/db.cljs (hunk 4 above)
Comment (madvas) [reply src/name_bazaar/ui/db.cljs]: okay

**Pull request closed without merging**